src/raftstore/store/worker/region.rs

```diff
@@ -326,3 +530,72 @@ impl Runnable<Task> for Runner {
         }
     }
 }
+
+impl RunnableWithTimer<Task, ()> for Runner {
+    fn on_timeout(&mut self, timer: &mut Timer<()>, _: ()) {
+        self.ctx.clean_timeout_ranges();
+        timer.add_task(Duration::from_millis(STALE_PEER_CHECK_INTERVAL), ());
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use std::thread;
+    use std::time::Duration;
+
+    use util::time;
+
+    use super::PendingDeleteRanges;
+
+    #[test]
+    fn test_pending_delete_ranges() {
+        let mut pending_delete_ranges = PendingDeleteRanges::default();
+        let delay = Duration::from_millis(100);
+        let id = 0;
+
+        let timeout = time::Instant::now() + delay;
+        pending_delete_ranges.insert(id, b"a".to_vec(), b"c".to_vec(), timeout);
```

Comment:
Can you reorganize these keys to make it more readable.

Reply:
I meant these `start_key` and `end_key` are not easy to see if they are overlap or not. It is better to use ranges like ['a' 'c'), ['d', 'f') and so on.